### PR TITLE
Added workflow to automatically rerun Python build on failure up to times

### PR DESCRIPTION
**Pull request by kevinmessiaen**
Added workflow to automatically rerun Python build on failure up to times

### CODE REVIEW

**Comment by alexcombessie** [comment]
@kevinmessiaen OK to merge, as tests are OK?

**Review by henchaves — APPROVED** [review]
LGTM, but we should have a second approval before merging 
cc @Hartorn

**Pull request merged**
